recipes/botan/all/conanfile.py

```diff
@@ -259,7 +263,7 @@ def _configure_cmd(self):
         if self.options.get_safe('fPIC', True):
             botan_extra_cxx_flags.append('-fPIC')
 
-        if tools.is_apple_os(self.settings.os):
+        if is_apple_os(self):
             if self.settings.get_safe('os.version'):
                 # Required, see https://github.com/conan-io/conan-center-index/pull/3456
                 macos_min_version = tools.apple_deployment_target_flag(self.settings.os,
```

Comment:
Unsure how this is best resolved. `apple_deployment_target_flag()` seems to be currently private because of a missing use case.

Don't know if we have here a use case, or if there is not a modern and better way. E.g. by using the `AutoToolsBuildEnvironment`, the only place there `apple_deployment_target_flag()` is called internally.

Reply:
Used `AutotoolsToolchain(self).apple_min_version_flag` as a workaround.